File: app/src/main/java/com/app/william/tribs/ui_board/StartScreenFragment.java
package com.app.william.tribs.ui_board;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.app.william.tribs.R;

/**
 * Created by William on 3/9/2016.
 */
public class StartScreenFragment extends Fragment {

    private StartScreenCallBacks mCallBacks;

    public interface StartScreenCallBacks {
        public void startGame();
    }

    private StartScreenCallBacks mDummyCallBacks = new StartScreenCallBacks() {
        @Override
        public void startGame() {
        }
    };

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.activity_starting_page, container, false);

        Button startBtn = (Button) view.findViewById(R.id.start_button);
        startBtn.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent e) {
                switch (e.getAction()) {
                    case MotionEvent.ACTION_DOWN:
                        v.setBackgroundDrawable(getResources().getDrawable(R.drawable.tile_white_pressed));
                        break;
                    case MotionEvent.ACTION_UP:
                            mCallBacks.startGame();
                        break;
                }
                return true;
            }
        });

        return view;
    }

    @Override
    public void onAttach(Context context){
        super.onAttach(context);

        if(context instanceof StartScreenCallBacks){
            mCallBacks = (StartScreenCallBacks) context;
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();

        mCallBacks = mDummyCallBacks;
    }
}

File: app/src/main/java/com/app/william/tribs/ui_board/StartScreenFragment.kt
package com.app.william.tribs.ui_board

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.MotionEvent
import android.view.View
import android.view.ViewGroup
import android.widget.Button

import com.app.william.tribs.R

/**
 * Created by William on 3/9/2016.
 */
class StartScreenFragment : Fragment() {

    private var mCallBacks: StartScreenCallBacks? = null

    interface StartScreenCallBacks {
        fun startGame()
    }

    private val mDummyCallBacks = object : StartScreenCallBacks {
        override fun startGame() {}
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.activity_starting_page, container, false)

        val startBtn = view.findViewById(R.id.start_button) as Button
        startBtn.setOnTouchListener { v, e ->
            when (e.action) {
                MotionEvent.ACTION_DOWN -> v.setBackgroundDrawable(resources.getDrawable(R.drawable.tile_white_pressed))
                MotionEvent.ACTION_UP -> mCallBacks!!.startGame()
            }
            true
        }

        return view
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)

        if (context is StartScreenCallBacks) {
            mCallBacks = context as StartScreenCallBacks?
        }
    }

    override fun onDetach() {
        super.onDetach()

        mCallBacks = mDummyCallBacks
    }
}
